File: python/src/apiz/_resources/tools.py
from __future__ import annotations

from .._http import AsyncHttpClient, SyncHttpClient
from .._types import ParseVideoResponse, TransferUrlResponse, TransferUrlType


class ToolsResource:
    def __init__(self, http: SyncHttpClient) -> None:
        self._http = http

    def parse_video(self, url: str) -> ParseVideoResponse:
        data = self._http.request(
            "POST", "/api/v3/tools/parse-video", body={"url": url}, auth=False
        )
        return ParseVideoResponse.model_validate(data)

    def transfer_url(self, url: str, *, type: TransferUrlType = "image") -> TransferUrlResponse:
        data = self._http.request(
            "POST",
            "/api/v3/tools/transfer-url",
            body={"url": url, "type": type},
            auth=False,
        )
        return TransferUrlResponse.model_validate(data)


class AsyncToolsResource:
    def __init__(self, http: AsyncHttpClient) -> None:
        self._http = http

    async def parse_video(self, url: str) -> ParseVideoResponse:
        data = await self._http.request(
            "POST", "/api/v3/tools/parse-video", body={"url": url}, auth=False
        )
        return ParseVideoResponse.model_validate(data)

    async def transfer_url(
        self, url: str, *, type: TransferUrlType = "image"
    ) -> TransferUrlResponse:
        data = await self._http.request(
            "POST",
            "/api/v3/tools/transfer-url",
            body={"url": url, "type": type},
            auth=False,
        )
        return TransferUrlResponse.model_validate(data)

File: packages/sdk/src/resources/tools.ts
import type { HttpClient } from "../http.js";
import type { ParseVideoResponse, TransferUrlResponse, TransferUrlType } from "../types.js";

export class ToolsResource {
  constructor(private readonly http: HttpClient) {}

  parseVideo(url: string): Promise<ParseVideoResponse> {
    return this.http.request<ParseVideoResponse>({
      method: "POST",
      path: "/api/v3/tools/parse-video",
      body: { url },
      auth: false,
    });
  }

  transferUrl(url: string, type: TransferUrlType = "image"): Promise<TransferUrlResponse> {
    return this.http.request<TransferUrlResponse>({
      method: "POST",
      path: "/api/v3/tools/transfer-url",
      body: { url, type },
      auth: false,
    });
  }
}
